.github/workflows/update-maintainers.yml

```diff
@@ -0,0 +1,245 @@
+name: Update maintainers list
+
+on:
+  push:
+    branches: ["main"]
+    paths: ["lib/maintainers.nix"]
+  schedule:
+    # Update every Monday at 9 AM UTC
+    - cron: "0 9 * * 1"
+  workflow_dispatch:
+    inputs:
+      create_pr:
+        description: "Create PR even if no changes"
+        required: false
+        default: false
+        type: boolean
+
+jobs:
+  update-maintainers:
+    runs-on: ubuntu-latest
+    # Creating PRs to update the generated maintainers file
+    permissions:
+      contents: write
+      pull-requests: write
+      issues: write
+    env:
+      pr_branch: update/maintainers-${{ github.ref_name }}
+    if: github.repository_owner == 'nix-community' || github.event_name == 'workflow_dispatch'
```

Comment:
Personally, I try to keep the conditional close to the start, usually immediately after the job name and/or `runs-on`.